src/gui/clips/MidiClipView.cpp

```diff
@@ -458,9 +458,74 @@ void MidiClipView::paintEvent( QPaintEvent * )
 	const int x_base = BORDER_WIDTH;
 
 	bool displayPattern = fixedClips() || (pixelsPerBar >= 96 && m_legacySEPattern);
-	// melody clip paint event
 	NoteVector const & noteCollection = m_clip->m_notes;
-	if( m_clip->m_clipType == MidiClip::MelodyClip && !noteCollection.empty() )
+
+	// Beat clip paint event (on BB Editor)
+	if (beatClip && displayPattern)
+	{
+		QPixmap stepon0;
+		QPixmap stepon200;
+		QPixmap stepoff;
+		QPixmap stepoffl;
+		const int steps = qMax( 1,
+					m_clip->m_steps );
+		const int w = width() - 2 * BORDER_WIDTH;
+
+		// scale step graphics to fit the beat clip length
+		stepon0 = s_stepBtnOn0->scaled( w / steps,
+					      s_stepBtnOn0->height(),
+					      Qt::IgnoreAspectRatio,
+					      Qt::SmoothTransformation );
+		stepon200 = s_stepBtnOn200->scaled( w / steps,
+					      s_stepBtnOn200->height(),
+					      Qt::IgnoreAspectRatio,
+					      Qt::SmoothTransformation );
+		stepoff = s_stepBtnOff->scaled( w / steps,
+						s_stepBtnOff->height(),
+						Qt::IgnoreAspectRatio,
+						Qt::SmoothTransformation );
+		stepoffl = s_stepBtnOffLight->scaled( w / steps,
+						s_stepBtnOffLight->height(),
+						Qt::IgnoreAspectRatio,
+						Qt::SmoothTransformation );
+
+		for( int it = 0; it < steps; it++ )	// go through all the steps in the beat clip
```

Comment:
Fixed formatting but ended up keeping the comment: It was already there since this is legacy code, it showed in the diffs because I moved places since I needed a different conditional flow. Thought the comments might help with refactoring of this code in the future.